### PR TITLE
http: add downstream remote reset response flag

### PR DESCRIPTION
**Pull request by botengyao**
Add response flag DownstreamRemoteReset when the http2 details are `http2.remote_reset` or `http2.remote_refuse` [here](https://github.com/envoyproxy/envoy/blob/8e93d16d433d3364c2b000dc9067ffc400e8f0d6/source/common/http/http2/codec_impl.cc#L1477-L1489)

This is similar as what router does for upstream
https://github.com/envoyproxy/envoy/blob/8e93d16d433d3364c2b000dc9067ffc400e8f0d6/source/common/router/router.cc#L1437-L1440

Commit Message:
Additional Description:
Risk Level: low
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by KBaichoo** [comment]
PTAL @wbpcode

File: test/common/http/conn_manager_impl_test_2.cc [hunk 1]
```diff
@@ -1769,6 +1769,87 @@ TEST_F(HttpConnectionManagerImplTest, ResetWithStoppedFilter) {
...
+    return Http::okStatus();
+  }));
+
+  Buffer::OwnedImpl fake_input("hello");
+  conn_manager_->onData(fake_input, false);
+  response_encoder_.stream_.setDetails("http2.remote_reset");
+  response_encoder_.stream_.resetStream(StreamResetReason::ConnectError);
+}
+
+TEST_F(HttpConnectionManagerImplTest, DownstreamRemoteResetRemoteReset) {
```
Comment (adisuissa) [inline test/common/http/conn_manager_impl_test_2.cc:1799]: nit: modify test-name (double RemoteReset) and add a one-line comment for each test

File: test/common/http/conn_manager_impl_test_2.cc [hunk 2]
```diff
@@ -1769,6 +1769,87 @@ TEST_F(HttpConnectionManagerImplTest, ResetWithStoppedFilter) {
...
+    return Http::okStatus();
+  }));
+
+  Buffer::OwnedImpl fake_input("hello");
+  conn_manager_->onData(fake_input, false);
+  response_encoder_.stream_.setDetails("http2.remote_reset");
+  response_encoder_.stream_.resetStream(StreamResetReason::RemoteReset);
+}
+
+TEST_F(HttpConnectionManagerImplTest, DownstreamRemoteResetRemoteRefusedStreamReset) {
```
Comment (adisuissa) [inline test/common/http/conn_manager_impl_test_2.cc:1826]: nit: same as above.

**Review by adisuissa — COMMENTED** [review]
LGTM, thanks!
Left a minor comment.
Please add a release note.

**Review by wbpcode — COMMENTED** [review]
LGTM except the comment from @adisuissa

**Comment by wbpcode** [comment]
Could you also check the CI failture?

File: test/common/http/conn_manager_impl_test_2.cc (hunk 1 above)
Comment (botengyao) [reply test/common/http/conn_manager_impl_test_2.cc]: Done, thanks!

File: test/common/http/conn_manager_impl_test_2.cc (hunk 2 above)
Comment (botengyao) [reply test/common/http/conn_manager_impl_test_2.cc]: Done

**Review by wbpcode — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**